ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/PlatformApplications.scala

```diff
@@ -110,10 +116,27 @@ object PlatformApplications {
         commandConfig = config.commandConfiguration,
         scenario = None,
         tlsConfig = None,
-        ledgerIdMode =
-          config.ledgerId.fold[LedgerIdMode](LedgerIdMode.Random)(LedgerIdMode.Predefined),
+        ledgerIdMode = config.ledgerId match {
+          case RequestedLedgerAPIMode.Static(id) =>
+            LedgerIdMode.Predefined(id)
+          case RequestedLedgerAPIMode.Dynamic() =>
+            LedgerIdMode.Random
+        },
         jdbcUrl = jdbcUrl
       )
     )
   }
 }
+
+sealed abstract class RequestedLedgerAPIMode extends Product with Serializable
```

Comment:
please remove this and use https://github.com/digital-asset/daml/blob/cc32ee5025125c5fc97b829796424fd21aab7e16/ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/config/LedgerIdMode.scala#L8 instead

Reply:
Semantics do not match. `LedgerIdMode.Random` is different from `RequestedLedgerAPIMode.Dynamic` for two reasons:
- `Dynamic` does not predefine if the ledger ID is random string or not, instead it just says that it is not part of the config;
- `Random` injects ledger ID configuration *into* the server under the fixture, which is impossible for `RemoteAPIProxy` fixture type I am about to implement.

Reply:
I see, would you consider merging the two ADTs in that case?

Reply:
I see, would you consider merging the two ADTs in that case?

Reply:
I am trying it now. It will move generation of random ledger ids into Sandbox itself, instead from the configuration level. I'll report back if it works.

Reply:
Done